app/build.gradle

```diff
@@ -164,3 +164,11 @@ task copyTextFiles(type: Copy) {
 }
 
 preBuild.dependsOn filterAbout, copyTextFiles
+
+allprojects {
+    gradle.projectsEvaluated {
+        tasks.withType(JavaCompile) {
+            options.compilerArgs << "-Xlint" << "-Xlint:-deprecation" << "-Xlint:-serial"
```

Comment:
I disable serialization due to the spurious warnings from extending `Exception` and disable deprecation due to calling several older APIs which are conditionally guarded.